packages/opentelemetry-types/src/resources/Resource.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** A Resource describes the entity for which a signal was collected. */
+export interface Resource {
+  /**
+   * A dictionary of labels with string keys and values that provide information
+   * about the entity.
+   */
+  readonly labels: { [key: string]: string };
```

Comment:
What other properties are expected to be on a `Resource`? Is it really necessary to have a dedicated class for this?

Reply:
From the spec, currently none. I can image having a dedicated class will allow us to add some later. I'm not sure but i think the right place to ask this kind of question should be on the spec repository since they made this specific choice

Reply:
I think it's possible that this was done because in Java there is a tendency to create a class for everything to allow extensions. Given the dynamic nature of JavaScript, this is usually unnecessary and I think this is an example of that.

Said that, maybe it should be acceptable to have a superset of the specification to provide more idiomatic APIs per language? I'm not sure how strict the current spec actually is.

Reply:
I wonder if we could still keep the `Resource` type but just make it an alias of the string object, so `type Resource = { [key: string]: string }`.

That way there is a nice nod to the specs but we don't have the extra indirection of another container object, and if we needed to change it in the future it might be easier to refactor.

Reply:
[OpenCensus Resource](https://github.com/census-instrumentation/opencensus-specs/blob/master/resource/Resource.md#resource-type) contains two properties: `type` and `labels`. I am not sure if will include same here in Otel. IMO current `Resource` interface will allow us to include more properties, if needed in future.

Reply:
Ah, I forgot about `type`. Is that part of the OpenTelemetry specs?

Reply:
Not 100% sure, @bogdandrutu question for you.

Reply:
This is what I found https://github.com/open-telemetry/opentelemetry-java/pull/145 (Remove type from the resource).

Reply:
Is the tradeoff around class vs. type here to use `new` or not when you want to add a new label?

Reply:
Yes, the advantage of an interface is essentially ease of use because you can just pass in an object literal like `{labels:{a: 'b', c: 'd'}}` and it will be a valid `Resource`. Whereas if it were a class, then yes, you would need to call `new` - it also means a bit more code bloat once it's compiled to JS, which is bad for the browser because you care a lot about code size there.